### PR TITLE
Fix SQL scripts

### PR DESCRIPTION
**Pull request by tkrille**
Fixes some issues with the SQL scripts. There's still 1 commit missing: splitting the SQL file for auth-server and resource-server as each might be using a dedicated database.

### CODE REVIEW

**Comment by tkrille** [comment]
not ready yet, forgot to adjust docs ... one moment, please

File: src/main/sql/admin_group.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+-- resource server: example group 'admin' for the addon-administration
```
Comment (dacrome) [inline src/main/sql/admin_group.sql]: actually it's not an example group, the 'admin' group is necessary to run this addon

File: src/main/sql/admin_group.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+INSERT INTO scim_meta (id, created, lastmodified, location, resourcetype, version)
+VALUES (2, '2011-10-10', '2011-10-10', NULL, 'Group', NULL);
```
Comment (dacrome) [inline src/main/sql/admin_group.sql]: we should make use of the [postgres](http://www.postgresql.org/docs/9.3/static/functions-datetime.html#FUNCTIONS-DATETIME-CURRENT) and [mysql](https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_current-date) CURRENT_DATE function

File: src/main/sql/example_data.sql [hunk 3]
```diff
@@ -1,8 +1,23 @@
...
+INSERT INTO osiam_client_scopes (id, scope) VALUES (20, 'GET');
+INSERT INTO osiam_client_scopes (id, scope) VALUES (20, 'POST');
+INSERT INTO osiam_client_scopes (id, scope) VALUES (20, 'PUT');
+INSERT INTO osiam_client_scopes (id, scope) VALUES (20, 'PATCH');
+INSERT INTO osiam_client_scopes (id, scope) VALUES (20, 'DELETE');
+INSERT INTO osiam_client_grants (id, grants) VALUES (20, 'authorization_code');
+
 -- resource server: example group 'admin' for the addon-administration
 
 INSERT INTO scim_meta (id, created, lastmodified, location, resourcetype, version)
```
Comment (dacrome) [inline src/main/sql/example_data.sql]: we should make use of the [postgres](http://www.postgresql.org/docs/9.3/static/functions-datetime.html#FUNCTIONS-DATETIME-CURRENT) and [mysql](https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_current-date) CURRENT_DATE function

File: src/main/sql/admin_group.sql (hunk 1 above)
Comment (tkrille) [inline src/main/sql/admin_group.sql]: good catch, thanks!

**Comment by tkrille** [comment]
all work done, ready to merge

**Comment by dacrome** [comment]
What about my comment?
> we should make use of the [postgres](http://www.postgresql.org/docs/9.3/static/functions-datetime.html#FUNCTIONS-DATETIME-CURRENT) and [mysql](https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_current-date) CURRENT_DATE function

**Comment by tkrille** [comment]
Sorry, cannot see it. But yes, why not? Seems as a viable way to pretend that the data is "fresh" :-)

**Comment by tkrille** [comment]
done

**Comment by dacrome** [comment]
I don't think so.

File: src/main/sql/admin_group.sql [hunk 4]
```diff
@@ -0,0 +1,18 @@
+--
+-- Group 'admin' for the addon-administration. Has to be imported in the
+-- database of the resource-server.
+--
+
+INSERT INTO scim_meta (id, created, lastmodified, location, resourcetype, version)
+VALUES (2, '2011-10-10', '2011-10-10', NULL, 'Group', NULL);
```
Comment (dacrome) [inline src/main/sql/admin_group.sql]: we should make use of the postgres and mysql CURRENT_DATE function

File: src/main/sql/example_data.sql [hunk 5]
```diff
@@ -1,20 +1,29 @@
...
+INSERT INTO osiam_client_scopes (id, scope) VALUES (20, 'PATCH');
+INSERT INTO osiam_client_scopes (id, scope) VALUES (20, 'DELETE');
+INSERT INTO osiam_client_grants (id, grants) VALUES (20, 'authorization_code');
 
 INSERT INTO scim_meta (id, created, lastmodified, location, resourcetype, version)
 VALUES (2, '2011-10-10', '2011-10-10', NULL, 'Group', NULL);
 
 INSERT INTO scim_id (internal_id, external_id, id, meta_id)
-VALUES (2, 'group_external_id', 'cef9452e-00a9-4cec-a086-d171374aabef', 2);
+VALUES (2, NULL, 'cef9452e-00a9-4cec-a086-d171374aabef', 2);
```
Comment (dacrome) [inline src/main/sql/example_data.sql]: You set this to NULL because I should not be searchable?

File: src/main/sql/example_data.sql [hunk 6]
```diff
@@ -1,20 +1,29 @@
...
+
+INSERT INTO osiam_client_scopes (id, scope) VALUES (20, 'GET');
+INSERT INTO osiam_client_scopes (id, scope) VALUES (20, 'POST');
+INSERT INTO osiam_client_scopes (id, scope) VALUES (20, 'PUT');
+INSERT INTO osiam_client_scopes (id, scope) VALUES (20, 'PATCH');
+INSERT INTO osiam_client_scopes (id, scope) VALUES (20, 'DELETE');
+INSERT INTO osiam_client_grants (id, grants) VALUES (20, 'authorization_code');
 
 INSERT INTO scim_meta (id, created, lastmodified, location, resourcetype, version)
 VALUES (2, '2011-10-10', '2011-10-10', NULL, 'Group', NULL);
```
Comment (dacrome) [inline src/main/sql/example_data.sql]: we should make use of the postgres and mysql CURRENT_DATE function

File: src/main/sql/example_data.sql (hunk 5 above)
Comment (tkrille) [inline src/main/sql/example_data.sql]: I don't know what this means, but AFAIC the external id should be empty, as it is controlled by the service consumer not the service provider.

**Pull request merged**